### PR TITLE
give much love to helpers

### PR DESCRIPTION
**Pull request by warlof**
drop non required css tweak
some code factor
no more roller coaster in layout

### CODE REVIEW

File: src/Http/Controllers/SeatGroupsController.php [hunk 1]
```diff
@@ -22,8 +22,9 @@ class SeatGroupsController extends Controller
      */
     public function index()
     {
-            return view('seatgroups::index')
-                ->with('seatgroups',SeatGroup::all());
+        $seatgroups = SeatGroup::all();
+
+        return view('seatgroups::index', compact('seatgroups'));
```
Comment (herpaderpaldent) [inline src/Http/Controllers/SeatGroupsController.php]: i see this pattern often in SeAT. What is the reason behind the compact? Does it save data volume? computing power?

File: src/Http/Controllers/SeatGroupsController.php [hunk 2]
```diff
@@ -142,11 +136,15 @@ public function update(Request $request, $id)
      */
     public function destroy($group_id)
     {
-        if(Auth::user()->hasRole('Superuser')){
+        if (auth()->user()->hasSuperUser()) {
```
Comment (herpaderpaldent) [inline src/Http/Controllers/SeatGroupsController.php]: What is the essential difference between `Auth::user()` and `auth()->user()` ?

File: src/Models/SeatGroup.php [hunk 3]
```diff
@@ -20,87 +20,93 @@
...
+                return $group->id;
+            })->toArray()) || auth()->user()->hasSuperUser();
     }
 
-    public function isAllowedToSeeSeatGroup(){
-
-        if(in_array(Auth::user()->group->main_character->corporation_id , $this->corporation->pluck('corporation_id')->toArray()) || Auth::user()->hasRole('seatgroups.edit')) {
+    public function isAllowedToSeeSeatGroup()
+    {
+        if (auth()->user()->hasSuperUser())
```
Comment (herpaderpaldent) [inline src/Models/SeatGroup.php]: i believe this if statement needs some curly parenthesis to return true?

File: src/Models/SeatGroup.php [hunk 4]
```diff
@@ -20,87 +20,93 @@
...
             return true;
-        }
-        return false;
 
+        return in_array(auth()->user()->group->main_character->corporation_id , $this->corporation->pluck('corporation_id')->toArray()) || auth()->user()->hasRole('seatgroups.edit');
```
Comment (herpaderpaldent) [inline src/Models/SeatGroup.php]: i'm glad you provide me with this ... makes it much more readable

File: src/Models/SeatGroup.php [hunk 5]
```diff
@@ -20,87 +20,93 @@
...
-                        return true;
-                }
-            }
-            if($this->type === "open"){
-                if(in_array(Auth::user()->group->main_character->corporation_id , $this->corporation->pluck('corporation_id')->toArray())){
-                    if(in_array(Auth::user()->group->id , $this->group->pluck('id')->toArray())){
+    public function isMember()
+    {
+        try {
+            switch ($this->type) {
```
Comment (herpaderpaldent) [inline src/Models/SeatGroup.php]: <3 ... didn't know this one but wow that helps

File: src/resources/views/seatgroups/create.blade.php [hunk 6]
```diff
@@ -81,14 +81,14 @@
...
 
-            @include('web::includes.javascript.id-to-name')
+    @include('web::includes.javascript.id-to-name')
 
-            <script>
-              $("#available_roles").select2({
-                placeholder: "{{ trans('web::seat.select_item_add') }}"
-              });
-            </script>
+    <script type="text/javascript">
```
Comment (herpaderpaldent) [inline src/resources/views/seatgroups/create.blade.php]: might be smth to do in the `edit.blade.php` of the roles in eveseat/web

File: src/Models/SeatGroup.php (hunk 3 above)
Comment (warlof) [reply src/Models/SeatGroup.php]: don't get the thing here. What do you mean ?

Comment (herpaderpaldent) [reply src/Models/SeatGroup.php]: ups should be: https://github.com/herpaderpaldent/seat-groups/pull/6/files/1b4a6bd1e92e45fad31a228a9fbb9263f4c06915#diff-2a009b09197595663d5e60c26dbff927R71

The thing is it reads right now:
```
public function isAllowedToSeeSeatGroup()
{
   if (auth()->user()->hasSuperUser())
      return true;

   {some other logic}
```
i would expect now that it return true everytime?

Comment (warlof) [reply src/Models/SeatGroup.php]: it will return true only if current user is super.

Comment (herpaderpaldent) [reply src/Models/SeatGroup.php]: i am no software engineer so i always thought one needs a closure after the if statement, which will be executed if the statement is true. Seems this is different in php. Thank you for your feedback!

Comment (warlof) [reply src/Models/SeatGroup.php]: code blocks are not mandatory. Short representation will always work, as longer as the thing to be done is on a single line.

That's working for `while`, `for`, `if`

**Pull request merged**